### PR TITLE
[improvement](External Catalog) Remove unnecessary conjuncts handling in External Catalog

### PR DESCRIPTION
**Pull request by zy-kkk**
In the previous FileScanNode, some parts that used conjuncts for predicate conversion were placed in the init phase. However, for the Nereids planner, pushing the filter down to the scan happens in the Translator, which means that the ScanNode can only get the complete conjuncts in the finalized phase. Therefore, in this PR, I have removed all conjuncts variables in External for the Nereids planner. They no longer need to store conjuncts themselves or add them to the ScanNode. Instead, all places in the ScanNode that use conjuncts should be moved to the finalized phase.

This refactor also fix a performance issue introduced from #40176
After introducing the change of generating SelectNode for consecutive projects or filters, FileScan still adds conjuncts too early in the init phase, resulting in the discovery of consecutive filters when the upper layer continues to translate, a selectnode was unexpectedly generated on the scannode, causing the project to be unable to prune the scannode columns. However, the Project node trims columns of SelectNode and ScanNode differently, which causes ScanNode to scan unnecessary columns.

My modification removes the addition of conjuncts in the scannode step, so that we can keep the structure from ScanNode to Project and achieve correct column trimming.

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/es/source/EsScanNode.java [hunk 1]
```diff
@@ -104,22 +104,22 @@ public EsScanNode(PlanNodeId id, TupleDescriptor desc, boolean esExternalTable)
     @Override
     public void init(Analyzer analyzer) throws UserException {
         super.init(analyzer);
-        buildQuery();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/es/source/EsScanNode.java:107]: do not modify legacy planner

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Review by morrySnow — COMMENTED** [review]
do we have external table partition prune test case in regression test?

**Comment by zy-kkk** [comment]
> do we have external table partition prune test case in regression test?

Yes, such as: test_hive_partition, test_hive_default_partition

**Pull request merged**